### PR TITLE
Overhaul AudioStreamPlayer's documentation

### LINKED ISSUE #76324: AudioStreamPlayer.get_stream_playback() returns null
### Godot version

v4.0.2.stable.official [7a0977ce2]

### System information

Windows 10, Forward+

### Issue description

Trying to get this project to run in Godot 4.  I imported it and ran the conversion process to upgrade to Godot 4.  

https://github.com/godotengine/godot-demo-projects/tree/3.5-9e68af3/audio/generator

When I try to run it, I get the error `Attempt to call function 'get_frames_available' in base 'null instance' on a null instance.`

### Steps to reproduce

Run the attached project.  When `$Player.get_stream_playback()` is called, it will return null which causes the rest of the demo to fail.

### Minimal reproduction project

[generator.zip](https://github.com/godotengine/godot/files/11298580/generator.zip)

### PR DESCRIPTION
**Pull request by Mickeon**
Closes https://github.com/godotengine/godot/issues/76324.
Documents https://github.com/godotengine/godot/issues/71122 's behavior.

Continuation of [similar past efforts](https://github.com/godotengine/godot/pulls?q=is%3Apr+author%3AMickeon+overhaul) with the same goal of refreshing the documentation up a notch.

**AudioStreamPlayer**'s reference smelled old. A lot of information is missing. Not everything has to be documented, of course. That would severely impact readability, by making the class _look_ more complicated than it actually is. This is a pretty common class for most users, and we have got to be careful.

PRs for **AudioStreamPlayer2D** and **AudioStreamPlayer3D** will come at a later date, after this one is merged.

--------
- The wording now matches other classes are documented a lot more closely;
 - A lot of missing information is added;
     - Unfortunately, unlike previous PRs, a bit of extra verbosity is necessary;
- Mention the ability to play **more than one sound** more clearly thorough the reference;
- Mention the importance of the `stream` property;
- Mention when the **`finished`** signal is emitted more explicitly;
    - The previous description failed to explain when this signal was **not** emitted, as much as it may seem like it should.
- Add note about **`MIX_TARGET_STEREO`** being the default mix target.

-------
### `get_playback_position`
- Imply there can be more than one **AudioStreamPlayback** (sound);
- Mention return value when no sounds are playing;
- Add note about https://github.com/godotengine/godot/issues/71122.

### `get_stream_playback`
- Mention the error when no sounds are played (https://github.com/godotengine/godot/issues/76324);
- Imply there can be more than one **AudioStreamPlayback**.

### `has_stream_playback`
- Simplify when this method returns `true`;

### `play`
- Be a little more explicit.
   - The original description quickly talks about the optional parameter. It doesn't make much sense to me, given that it's entirely optional.

### `seek`
- Be a little more explicit;
- Imply there can be more than one **AudioStreamPlayback** (sound).

### `stop`
- Be a little more explicit.
    - ###### "Stops all audio" is it like... my entire system becomes deaf or something?

-------
### `autoplay`
- Be a little more explicit.
    - The previous description was fairly awkward.

### `bus`
- Simplified the note **considerably**.
    - What a mouthful... More experienced users already know the bus layout can be changed. It's also overwhelming information for **AudioStreamPlayer**. It's only necessary to describe the fall back behavior.

### `max_polyphony`
- Be a little more explicit.

### `mix_target`
- Shuffled wording around to be clearer;
- Mention how to get the configured speaker mode yourself.

### `pitch_scale`
- Add example on how this property affects the audio.

### `playing`
- Describe what happens when setting this property.

### `stream`
- Describe what happens when setting this property.

### `stream_paused`
- Imply there can be more than one **AudioStreamPlayback** (sound).
- Mention that this property is changed automatically on a bunch of occasions.
    - I can't find it, but I believe there was an issue about this.

### `volume_db`
- Be a little more explicit.
    - The concept of audio volume is so widely understood it probably doesn't need further explanations. (such as "_Higher values make the sound louder_"). This is fine.
- Add note about the linear/db conversion functions.
----------------

I don't think this PR is fully complete. The leading description needs a bit more to it. But, I wanted to express this PR's intentions as soon as possible. **Feedback is very, very welcome**.

In particular, I attempted to reduce verbosity by replacing "_audio_" when referring to the individual **AudioStreamPlayback**s with "_sounds_". Saying "_playback_" all the time does not quite convince me as it's repetitive, and is not something that the common user will ever have to think about, so it's a puzzling situation.

### CODE REVIEW

File: doc/classes/AudioStreamPlayer.xml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AudioStreamPlayer" inherits="Node" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
-		Plays back audio non-positionally.
+		Plays audio without a position.
 	</brief_description>
 	<description>
-		Plays an audio stream non-positionally.
-		To play audio positionally, use [AudioStreamPlayer2D] or [AudioStreamPlayer3D] instead of [AudioStreamPlayer].
+		The [AudioStreamPlayer] plays an audio stream non-positionally. It is ideal for user interfaces, menus, or background music.
```
[the remark this answers is not in the document]
Comment (Mickeon) [reply doc/classes/AudioStreamPlayer.xml:7]: I don't know. Why and/or why not adding the article to start the sentence? Either seem fine but...

[the remark this answers is not in the document]
Comment (MewPurPur) [reply doc/classes/AudioStreamPlayer.xml:7]: "The" suggests a specific AudioStreamPlayer, removing "the" refers to the class.

[the remark this answers is not in the document]
Comment (MewPurPur) [reply doc/classes/AudioStreamPlayer.xml:7]: That's also fine, but I think it's best without either

File: doc/classes/AudioStreamPlayer.xml [hunk 2]
```diff
@@ -19,82 +20,85 @@
...
 			<description>
-				Returns the [AudioStreamPlayback] object associated with this [AudioStreamPlayer].
+				Returns the latest [AudioStreamPlayback] of this node, usually the most recently created by [method play]. If no sounds are playing, this method fails and returns an empty playback.
 			</description>
 		</method>
 		<method name="has_stream_playback">
 			<return type="bool" />
 			<description>
-				Returns whether the [AudioStreamPlayer] can return the [AudioStreamPlayback] object or not.
+				Returns [code]true[/code] if any sound is currently playing. See also [member playing] and [method get_stream_playback].
```
Comment (RedMser) [inline doc/classes/AudioStreamPlayer.xml:36]: Should be clarified. `has_stream_playback` returns `true` is a stream has been started, but then `stream_paused = true` was set. `playing` returns `false` in this case.

File: doc/classes/AudioStreamPlayer.xml [hunk 3]
```diff
@@ -19,82 +20,85 @@
...
+			The target bus name. All sounds from this node will be playing on this bus.
+			[b]Note:[/b] At runtime, if no bus with the given name exists, all sounds will fall back on [code]"Master"[/code]. See also [method AudioServer.get_bus_name].
 		</member>
 		<member name="max_polyphony" type="int" setter="set_max_polyphony" getter="get_max_polyphony" default="1">
-			The maximum number of sounds this node can play at the same time. Playing additional sounds after this value is reached will cut off the oldest sounds.
+			The maximum number of sounds this node can play at the same time. Calling [method play] after this value is reached will cut off the oldest sounds.
 		</member>
 		<member name="mix_target" type="int" setter="set_mix_target" getter="get_mix_target" enum="AudioStreamPlayer.MixTarget" default="0">
-			If the audio configuration has more than two speakers, this sets the target channels. See [enum MixTarget] constants.
+			The mix target channels, as one of the [enum MixTarget] constants. Has no effect when two speakers or less are detected (see [method AudioServer.get_speaker_mode]).
```
Comment (RedMser) [inline doc/classes/AudioStreamPlayer.xml:72]: The documentation note about "two speakers or less" is part of `SpeakerMode` enum, so the reference should point there instead.

**Comment by ronyeh** [comment]
Looks good. Improving documentation is very valuable work, and helps new developers!

**Comment by Mickeon** [comment]
Rebased after minor conflict. Would be nice for this to be looked at sooner than later, so that both AudioStreamPlayer2D and AudioStreamPlayer3D may receive a similar treatment

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**